### ISSUE TITLE
[BUG] Widgets using tag : 'someNaturallyFocusableTagName' erroneously end up with role="presentation"

### ISSUE THREAD

**Issue by ExtAnimal** [bryntum/support#13440]
See the `[data-ref="supportButton"]` element below:

<img width="918" height="728" alt="Image" src="https://github.com/user-attachments/assets/1fcdd4b6-accb-4201-af57-14ad721f0eb8" />

The issue is that `Widget#updateRole` applies Widget's default role of "presentation" by default.

If there is no configured role and the element is naturally focusable with its own intrinsic role (eg an `<a>`), then "presentation" is invalid.

A suggested solution in `Widget.js` is 

```
    updateRole(role) {
        const { ariaElement } = this;

        // Do not apply the default "presentation" role if the element is focusable,
        // and not a plain DIV, since that would override the element's default role.
        if (role && !(ariaElement?.nodeName !== 'DIV' && !this.initialConfig.role && DomHelper.isFocusable(ariaElement, true))) {
            ariaElement?.setAttribute('role', role);
        }
        else {
            ariaElement?.removeAttribute('role');
        }
    }
```

Also, in `Button#compose`, the type should be set thusly:

```
type  : href ? undefined : behaviorType,
```
